test/test_cpp_extensions_aot.py

```diff
@@ -133,6 +147,100 @@ def test_cuda_dlink_libs(self):
         test = cuda_dlink.add(a, b)
         self.assertEqual(test, ref)
 
+
+class TestPybindTypeCasters(common.TestCase):
```

Comment:
Wasn't sure if this was the right place for this test, but it was the most convenient since the framework for building and importing the cpp_extension is already in place.